app/services/userAlbums.js

```diff
@@ -0,0 +1,14 @@
+const { UserAlbum } = require('../models');
+const logger = require('../logger');
+const errors = require('../errors');
+
+exports.buyAlbum = data =>
+  UserAlbum.create(data).catch(error => {
+    logger.error(error);
+    throw errors.databaseError(error.message);
+  });
```

Comment:
How do you handle the duplicate album here?

Reply:
It has a unique constraint

Reply:
So it cannot possibly create a duplicate

Reply:
But you notify that to the user as a database error? 5XX, instead of 4XX (like 409 conflict)

Reply:
409 would be more appropriate